client/src/modules/authorization/components/PasswordRecoveryForm.jsx

```diff
@@ -0,0 +1,227 @@
+import React, { PureComponent } from 'react';
+import { FormattedMessage, injectIntl } from 'react-intl';
+import { withRouter } from 'react-router-dom';
+import { connect } from 'react-redux';
+import validator from 'validator';
+import _flowRight from 'lodash/flowRight';
+import _debounce from 'lodash/debounce';
+import _isEmpty from 'lodash/isEmpty';
+import _trim from 'lodash/trim';
+
+import { RouterMatchPropType, IntlPropType } from 'common/constants/propTypes';
+import { updatePassword } from '../model/actions';
+import ValidationInput from './ValidationInput';
+import PendingButton from '../../../common/components/PendingButton';
+
+class PasswordRecoveryForm extends PureComponent {
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      errors: {
+        comparePasswordsError: '',
+        passwordError: '',
+        passwordUpdateError: false
+      },
+      password: '',
+      passwordConfirmation: '',
+      passwordConfirmationSuccess: false,
+      passwordSuccess: false,
+      pending: false
+    };
+
+    this.debouncedChange = undefined;
+  }
+
+  handlePasswordChange = event => {
+    const {
+      target: { value }
+    } = event;
+
+    this.debouncedChange = _debounce(() => this.validatePassword(value), 500);
```

Comment:
this should be declared in constructor and the debounced function should be used

Reply:
why it should be in the constructor?

Reply:
debounced function should be created only once and then used - currently you are creating new debounced function on every `handlePasswordChange` call. see https://lodash.com/docs/4.17.11#debounce